src/inspector/worker_inspector.h

```diff
@@ -0,0 +1,97 @@
+#ifndef SRC_INSPECTOR_WORKER_INSPECTOR_H_
+#define SRC_INSPECTOR_WORKER_INSPECTOR_H_
+
+#if !HAVE_INSPECTOR
+#error("This header can only be used when inspector is enabled")
+#endif
+
+#include <memory>
+#include <string>
+#include <unordered_map>
+#include <unordered_set>
+
+namespace node {
+namespace inspector {
+class MainThreadHandle;
+class WorkerManager;
+
+class WorkerDelegate {
+ public:
+  virtual void WorkerCreated(const std::string& title,
+                             const std::string& url,
+                             bool waiting,
+                             std::shared_ptr<MainThreadHandle> worker) = 0;
+};
+
+class WorkerManagerEventHandle {
+ public:
+  explicit WorkerManagerEventHandle(std::shared_ptr<WorkerManager> manager,
+                                    int id)
+                                    : manager_(manager), id_(id) {}
+  void SetWaitOnStart(bool wait_on_start);
+  ~WorkerManagerEventHandle();
+
+ private:
+  std::shared_ptr<WorkerManager> manager_;
+  int id_;
+};
+
+struct WorkerInfo {
+  WorkerInfo(const std::string& target_title,
+             const std::string& target_url,
+             std::shared_ptr<MainThreadHandle> target_interface)
+             : title(target_title),
+               url(target_url),
+               interface(target_interface) {}
+  std::string title;
+  std::string url;
+  std::shared_ptr<MainThreadHandle> interface;
+};
+
+class ParentInspectorHandle {
+ public:
+  ParentInspectorHandle(int id, const std::string& url,
+                        std::shared_ptr<MainThreadHandle> parent_thread,
+                        bool wait_for_connect);
+  ~ParentInspectorHandle();
+  void WorkerStarted(std::shared_ptr<MainThreadHandle> target_interface,
+                     bool waiting);
+  bool WaitForConnect() {
+    return wait_;
+  }
+ private:
```

Comment:
Style nit: blank line before `private:` (I think the linter should complain about this?)